src/actions/motion.ts

```diff
@@ -1841,12 +1903,71 @@ abstract class MoveTagMatch extends BaseMovement {
         failed: true,
       };
     }
+    vimState.cursorStartPosition = startPosition;
     return {
       start: startPosition,
       stop: endPosition.getLeftThroughLineBreaks(true),
     };
   }
 
+  public async execActionWithCount(
+    position: Position,
+    vimState: VimState,
+    count: number
+  ): Promise<Position | IMovement> {
+    let recordedState = vimState.recordedState;
+    let result: Position | IMovement = new Position(0, 0); // bogus init to satisfy typechecker
+
+    if (count < 1) {
```

Comment:
I know we've had this pattern but we should really move this sort of check somewhere outside this function (on a different PR).

Reply:
I was thinking of abstracting more of the code in this function to private methods. This would probably allow us to add a couple of classes between `BaseMovement` and its subclasses and reduce duplication. That might also help in the future to refactor how count-prefixes are handled before reaching `BaseMovement`.

Reply:
Were you thinking of doing that in this PR or sometime later?

Reply:
I'll try to cleanup some of the changes in this PR and sprinkle in some more comments. If you have any other concerns or suggestions I'd like to organize what I need to do so I can plan some time to get to it.